var/spack/repos/builtin/packages/hdfview/package.py

```diff
@@ -55,4 +56,5 @@ def install(self, spec, prefix):
     def setup_build_environment(self, env):
         env.set("HDF5LIBS", self.spec["hdf5"].prefix)
         env.set("HDFLIBS", self.spec["hdf"].prefix)
+        env.set("JAVA_HOME", self.spec["java"].prefix)
```

Comment:
IIRC you mentioned this was built against `openjdk`, would you be able to try removing this line and run `spack build-env hdfview bash` in your environment and see what `JAVA_HOME` refers to (or if it isn't defined, mention that)?

The `openjdk` package is supposed to set `JAVA_HOME`, but it might be that it is setting it to the wrong thing (in some cases it does not set to `spec['java'].prefix`).

Reply:
Hi @scheibelp I wanted to do this, but then got caught in many other emergencies and it looks like I will be so for at least the next week or ten days. I hope to eventually get to this. Thanks for your patience.